src/skin/legacyskinparser.cpp

```diff
@@ -1789,6 +1789,31 @@ QString LegacySkinParser::getStyleFromNode(const QDomNode& node) {
             style = QString::fromLocal8Bit(fileBytes.constData(),
                                            fileBytes.length());
         }
+
+        double scaleFactor = m_pContext->getScaleFactor();
+        if (scaleFactor >= 3) {
+            // Try to load with @3x suffix
+            QFileInfo info(file);
+            QString strNewName = info.path() + "/" + info.baseName() + "@3x."
+                    + info.completeSuffix();
+            QFile file(strNewName);
+            if (file.open(QIODevice::ReadOnly)) {
+                QByteArray fileBytes = file.readAll();
+                style.prepend(QString::fromLocal8Bit(fileBytes.constData(),
+                                               fileBytes.length()));
+            }
+        } else if (scaleFactor >= 2) {
```

Comment:
I have adopted this form the Gnome/Unity Scaling slider. 
It is one floating point sider for the fonts and widgets. The widgets are doubled once you reach the 2.0 threshold. 
This means, once we are at Qt 5, the native widgets will also scale at this point, Which may require some style tweaks, at this threshold. 

I will add a comment.